onnxscript/rewriter/custom_ops/llm_rule_sets_cuda.py

```diff
@@ -0,0 +1,495 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+from __future__ import annotations
+
+import onnx
+
+import onnxscript.ir as ir
+import onnxscript.rewriter.pattern as orp
+
+op = orp.onnxop
+
+
+class _CombineBinary(orp.RewriteRuleAsClass):
+    @classmethod
+    def _same_shape(
+        cls, sh1: tuple[int, ...], sh2: tuple[int, ...], broadcast: bool = False
+    ) -> bool:
+        if broadcast:
+            if len(sh1) != len(sh2):
+                rk = max(len(sh1), len(sh2))
+                sh1 = (1,) * (rk - len(sh1)) + sh1
+                sh2 = (1,) * (rk - len(sh2)) + sh2
+            allow_one1 = True
+            allow_one2 = True
+            for a, b in zip(sh1, sh2):
+                if a == b:
+                    if a != 1:
+                        allow_one1 = False
+                    if b != 1:
+                        allow_one2 = False
+                    continue
+                if a == 1 and allow_one1:
+                    allow_one2 = False
+                    continue
+                if b == 1 and allow_one2:
+                    allow_one1 = False
+                    continue
+                return False
+            return True
+        return sh1 == sh2
+
+    @classmethod
+    def check(cls, context, x, y, z) -> bool:
+        if x.shape is None or y.shape is None or z.shape is None:
+            return False
+        return cls._same_shape(x.shape, y.shape, broadcast=True) and cls._same_shape(
+            y.shape, z.shape, broadcast=True
+        )
+
+
+class CombinedAddAdd1(_CombineBinary):
+    @classmethod
+    def pattern(cls, op, x, y, z):
+        return op.Add(x, op.Add(y, z))
+
+    @classmethod
+    def rewrite(cls, op, x, y, z):
+        return op.AddAdd(x, y, z, domain="ai.onnx.contrib")
```

Comment:
What is the `ai.onnx.contrib` domain?

Reply:
kernel implemented in onnxruntime-extensions

Reply:
Technically, the ONNX org owns the `ai.onnx` prefix. It is probably not a great idea for microsoft to use that without some approval from some ONNX authority ... not sure why they didn't use `com.microsoft.contrib` ?

Reply:
Idea being to avoid name conflicts ... e.g., if AMD, Nvidia, etc. start doing the same thing, it would be a problem.